File: demos/solar-system/reference/Circle.java
//import java.awt.*;
public class Circle extends Shape
{
    private double myDiameter;
    
//Constructors
    
      public Circle()
    {
        super(0,0);
        myDiameter = 10;
    }
    public Circle(Point p, double diam)
    {
        super(p.x,p.y);
        myDiameter = diam;
    }
    
//Accessors
    public double diameter()
    { return myDiameter; }
    
    public double radius()
    { return diameter()/2.0; }
    
    public double height()
    { return diameter(); }
    
    public double width()
    { return diameter(); }
    
    public void moveTo(Point p)
    {
        setXcenter(p.x);
        setYcenter(p.y);
    }
    public void move(Vector v)
    {
        Vector t = (Vector)v.clone();
        t.translate(center());//setStart
        moveTo(t.endPt());
    }
    
//Actions
    public void show(java.awt.Graphics g)
    {
        //transform the view frame
        int d = (int)(diameter());// /(6*s[si]));
        //double s = 1;
        //Point p= new Point(0,0);
        //int d = (int)(diameter()/s);
        double x, y;
        x=Xcenter();
        y=Ycenter();
        /*x+=p.x;
        y+=p.y;
        x*=s[si];
        y*=s[si];
        x+=p.x;
        y+=p.y;*/
        g.fillOval((int)x,(int)y,d,d);
        
    }
}

File: demos/solar-system/circle.js
class Circle extends Shape
{
    // private double myDiameter;

//Constructors

    constructor(x)
    {
        if (x != null) {
            throw new Error('CIrcle has default constructor');
        }
        super();
        super.initXY(0,0);
        this.myDiameter = 10;
    }

    // p - Point
    // diam - double
    static fromPointDiameter(p, diam)
    {
        let c = new Circle();
        c.setXcenter(p.x);
        c.setYcenter(p.y);
        c.myDiameter = diam;
        return c;
    }

    initPointDiameter(p, diam)
    {
        this.setXcenter(p.x);
        this.setYcenter(p.y);
        this.myDiameter = diam;
    }
//Accessors
    // returns double
    diameter()
    { return this.myDiameter; }

    // returns double
    radius()
    { return this.diameter()/2.0; }

    // returns double
    height()
    { return this.diameter(); }

    // returns double
    width()
    { return this.diameter(); }
    // p - Point
    moveTo(p)
    {
        this.setXcenter(p.x);
        this.setYcenter(p.y);
    }
    // v - Vector
    move(v)
    {
        let t = v.clone();
        t.translate(this.center());//setStart
        this.moveTo(t.endPt());
    }

//Actions
    show(ctx)
    {
        ctx.fillStyle=this.color();
        //transform the view frame
        let d = this.diameter();// /(6*s[si]));
        let x, y;
        x=this.Xcenter();
        y=this.Ycenter();
        //TODO
        // g.fillOval((int)x,(int)y,d,d);
        ctx.arc(x, y, 0.5*d, 0, 2 * Math.PI);

    }
}